### DISCUSSION TITLE
Or Conditional on Collections Tag

### DISCUSSION THREAD

**Discussion by mgtcampbell** [statamic/cms#7692] [Q&A] [answered]
Hi there,

I have a situation where I want to pull entries from an "events" collection for a particular location, so I want to filter the collection by something like `:locations:contains="id"` - so it only shows events with locations that include the id of the page I'm serving (the location page).

However, I also want to show an events that have an event_type of "webinar", since webinars can be accessed from any location.

Essentially I'm looking for something like this:

`{{ collection:events status:is="published" { :locations:contains="id" OR event_type == 'webinar' } as="events" }}`

However I can't seem to get it to work that it only shows events that meet one or the other of those conditions.

I have also tried just getting the entire events collection, and then immediately inside the collections tag to add another condition like so:

```
{{ collection:events status:is="published" as="events" }}
  {{ events }}
    {{ if  locations | contains("id") || event_type == 'webinar' }}
...
```

However in this scenario I then get an error `Passed object must have a __toString method`

Locations is just an array of IDs so I am unsure why the contains modifier is throwing an error.

Any help appreciated.

Cheers

**Comment by ryanmitchell** [comment, marked as answer]
Did you look into [Query Scopes](https://statamic.dev/tags/collection#custom-query-scopes) ?

**Reply by mgtcampbell** [reply]
Thanks for the tip - I'll take a look!

**Reply by mgtcampbell** [reply]
I was able to make something work by adding a Query Scope.

I couldn't figure out how exactly how to convert `:locations:contains="id"` into a part of the Query Scope, so I just left that on the Collections tag and then adjusted the Query Scope to give me what I want after that was applied.

So this was my collection tag:

`{{ collection:events :locations:contains="id" query_scope="events_by_location" as="events" }}`

And then this was my Query Scope

```
class EventsByLocation extends Scope
{
    /**
     * Apply the scope.
     *
     * @param \Statamic\Query\Builder $query
     * @param array $values
     * @return void
     */
    public function apply($query, $values)
    {

        $query->orWhere('event_type', 'webinar')->where('status', 'published');

    }
}
```

**Comment by ryanmitchell** [comment]
> I couldn't figure out how exactly how to convert :locations:contains="id" into a part of the Query Scope, so I just left that on the Collections tag and then adjusted the Query Scope to give me what I want after that was applied.

```php
$query->whereJsonContains('locations', $values['location_id'])
```

```
{{ collection:events :location_id="id" query_scope="events_by_location" as="events" }}
```

**Comment by mgtcampbell** [comment]
Awesome thanks!

On Wed, 15 Mar 2023 at 6:35 pm, Ryan Mitchell ***@***.***>
wrote:

> I couldn't figure out how exactly how to convert :locations:contains="id"
> into a part of the Query Scope, so I just left that on the Collections tag
> and then adjusted the Query Scope to give me what I want after that was
> applied.
>
> $query->whereJsonContains('locations', $values['location_id'])
>
> {{ collection:events :location_id="id" query_scope="events_by_location" as="events" }}
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/statamic/cms/discussions/7692#discussioncomment-5318676>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA4VWDN5HVOVHBDPKPLN5JDW4F5NZANCNFSM6AAAAAAVZZQVYU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 
[image: Where brands are built.] <https://htmlsig.com/t/000001FQ5N7T>

Mark Campbell0431 626 082

Where brands are built.
ply.studio
